pkg/server/chaosd/disk.go

```diff
@@ -0,0 +1,172 @@
+
+func (s *Server) DiskPayload(payload *core.DiskCommand) (uid string, err error) {
+	uid = uuid.New().String()
+
+	if err = s.exp.Set(context.Background(), &core.Experiment{
+		Uid:            uid,
+		Status:         core.Created,
+		Kind:           core.DiskAttack,
+		Action:         payload.Action,
+		RecoverCommand: payload.String(),
+	}); err != nil {
+		err = errors.WithStack(err)
+		return
+	}
+	defer func() {
+		if err != nil {
+			if err := s.exp.Update(context.Background(), uid, core.Error, err.Error(), payload.String()); err != nil {
+				log.Error("failed to update experiment", zap.Error(err))
+			}
+			return
+		}
+		if err := s.exp.Update(context.Background(), uid, core.Success, "", payload.String()); err != nil {
+			log.Error("failed to update experiment", zap.Error(err))
+		}
+	}()
+
+	switch payload.Action {
+	case core.DiskWritePayloadAction:
+		if payload.Path == "" {
+			payload.Path = "/dev/null"
+		}
+		cmd := exec.Command("bash", "-c", fmt.Sprintf(DDWritePayloadCommand, payload.Path, "1M", strconv.FormatUint(payload.Size, 10)))
+		output, err := cmd.CombinedOutput()
+
+		if err != nil {
+			log.Error(string(output), zap.Error(err))
+		} else {
+			log.Info(string(output))
+		}
+		return uid, err
+	case core.DiskReadPayloadAction:
+		if payload.Path == "" {
+			err := errors.Errorf("empty read payload path")
+			log.Error(fmt.Sprintf("payload action: %s", payload.Action), zap.Error(err))
+			return uid, err
+		}
+		cmd := exec.Command("bash", "-c", fmt.Sprintf(DDReadPayloadCommand, payload.Path, "1M", strconv.FormatUint(payload.Size, 10)))
+		output, err := cmd.CombinedOutput()
+
+		if err != nil {
+			log.Error(string(output), zap.Error(err))
+		} else {
+			log.Info(string(output))
+		}
+		return uid, err
+	default:
+		err := errors.Errorf("invalid payload action")
+		log.Error(fmt.Sprintf("payload action: %s", payload.Action), zap.Error(err))
+		return uid, err
+	}
+}
+
+const DDFillCommand = "dd if=/dev/zero of=%s bs=%s count=%s iflag=fullblock"
+const DDFallocateCommand = "fallocate -l %sM %s"
+
+func (s *Server) DiskFill(fill *core.DiskCommand) (uid string, err error) {
+	uid = uuid.New().String()
+
+	if err = s.exp.Set(context.Background(), &core.Experiment{
+		Uid:            uid,
+		Status:         core.Created,
+		Kind:           core.DiskAttack,
+		Action:         fill.Action,
+		RecoverCommand: fill.String(),
+	}); err != nil {
+		err = errors.WithStack(err)
+		return
+	}
+	defer func() {
+		if err != nil {
+			if err := s.exp.Update(context.Background(), uid, core.Error, err.Error(), fill.String()); err != nil {
+				log.Error("failed to update experiment", zap.Error(err))
+			}
+			return
+		}
+		if err := s.exp.Update(context.Background(), uid, core.Success, "", fill.String()); err != nil {
+			log.Error("failed to update experiment", zap.Error(err))
+		}
+	}()
+
+	if fill.Path == "" {
+		for {
+			tempFilename := rand.String(128) + "temp-data"
```

Comment:
how about using [TempFile](https://golang.org/pkg/io/ioutil/#TempFile)?

Reply:
OK, I agree with you.